subprojects/packagefiles/google-highway/meson.build

```diff
@@ -0,0 +1,155 @@
+project(
+  'highway',
+  'cpp',
+  version: '1.2.0',
+  license: 'Apache-2.0',
+)
+
+highway_sources = files(
+    'hwy/abort.cc',
+    'hwy/aligned_allocator.cc',
+    'hwy/nanobenchmark.cc',
+    'hwy/per_target.cc',
+    'hwy/print.cc',
+    'hwy/targets.cc',
+    'hwy/timer.cc',
+)
+
+if host_machine.system() == 'windows'
+  add_project_arguments('-DNOMINMAX', language: 'cpp')
+endif
+is_shared = (get_option('default_library') != 'static')
+args = is_shared ? ['-DHWY_SHARED_DEFINE', '-Dhwy_EXPORTS'] : ['-DHWY_STATIC_DEFINE']
+
+lib_highway = library(
+    'highway',
+    highway_sources,
+    cpp_args: args,
+    gnu_symbol_visibility: 'hidden',
+)
+
+inc = include_directories('.')
+
+google_highway_dep = declare_dependency(
+    link_with: lib_highway,
+    include_directories: inc,
+)
```

Comment:
```meson
meson.override_dependency('hwy', google_highway_dep)
```

Reply:
Is this the general pattern to always use `meson.override_dependency(...)` for internally declared dependencies? I assume this prevents meson from finding the dependency elsewhere on the system?

Reply:
This is only used for dependencies that you want to export, so in this case `hwy`, but if you have an internally used dependency you don't want to export (say if you wrapped the test helper library in a dependency), then you wouldn't do this.

Yes, it short circuits `dependency('hwy')` to return this dependency.